src/components/pages/tours/TourDetail.tsx

```diff
@@ -75,9 +85,16 @@ const TourDetail = () => {
       </Grid>
       <Divider/>
       {isOnline &&
-        <MapWrapper>
-          <WayPointMarkerLayer features={[tour.startLocation, tour.endLocation]}/>
-        </MapWrapper>
+          <>
+              <MapWrapper>
+                  <WayPointMarkerLayer features={[new TourPoint(tour.startLocation), new TourPoint(tour.endLocation)]}/>
```

Comment:
I would move this to a function so it is more readable. Maybe const getWayPointMarkerFeatures= () {}?

Reply:
I get your point and I'll add this. However, it's a wee bit ugly because we have to make one more `if (tour)` assertion.